src/shredder/reed_solomon.rs

```diff
@@ -4,35 +4,52 @@
 //! Implements Reed-Solomon shreding and deshreding.
 
 use reed_solomon_simd as rs;
+use static_assertions::const_assert;
+use thiserror::Error;
 
 use super::{
-    CodingShred, DATA_SHREDS, DataShred, MAX_DATA_PER_SLICE, Shred, ShredPayload, ShredPayloadType,
-    SliceHeader, SlicePayload, TOTAL_SHREDS,
+    CodingShred, DATA_SHREDS, DataShred, MAX_DATA_PER_SLICE, MAX_DATA_PER_SLICE_AFTER_PADDING,
+    Shred, ShredPayload, ShredPayloadType, SliceHeader, SlicePayload, TOTAL_SHREDS,
 };
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Error)]
 pub(super) enum ReedSolomonShredError {
+    #[error("too much data for slice")]
     TooMuchData,
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Error)]
 pub(super) enum ReedSolomonDeshredError {
+    #[error("not enough shreds to reconstruct")]
     NotEnoughShreds,
+    #[error("more shreds than expected")]
     TooManyShreds,
+    #[error("too much data for slice")]
     TooMuchData,
 }
 
 /// Splits the given slice into `num_data` data shreds, then generates
 /// `num_coding` additional Reed-Solomon coding shreds.
 pub(super) fn reed_solomon_shred(
     header: SliceHeader,
-    payload: SlicePayload,
+    mut payload: SlicePayload,
     num_data: usize,
     num_coding: usize,
 ) -> Result<(Vec<DataShred>, Vec<CodingShred>), ReedSolomonShredError> {
     if payload.len() > MAX_DATA_PER_SLICE {
         return Err(ReedSolomonShredError::TooMuchData);
     }
 
-    let shred_bytes = payload.len() / DATA_SHREDS;
+    // add padding
+    // TODO: use a padding scheme that can support larger slices
+    const_assert!(2 * DATA_SHREDS < 256);
+    let padding_bytes = (2 * DATA_SHREDS - payload.len() % (2 * DATA_SHREDS)) as u8;
+    payload
+        .data
+        .resize(payload.len() + padding_bytes as usize, padding_bytes);
+    assert!(payload.len() <= MAX_DATA_PER_SLICE_AFTER_PADDING);
+
+    let shred_bytes = payload.len().div_ceil(DATA_SHREDS);
```

Comment:
I am fine with this code.  But just to improve my understanding, I believe that payload.len() should be a multiple of DATA_SHREDS at this point in time, no?